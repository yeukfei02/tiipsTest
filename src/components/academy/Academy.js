import React from 'react';
import { StyleSheet, View, Text } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});

function Academy() {
  return (
    <View style={styles.container}>
      <Text>Academy</Text>
    </View>
  );
}

export default Academy;
