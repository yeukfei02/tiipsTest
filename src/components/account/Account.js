import React from 'react';
import { StyleSheet, View, Text } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});

function Account() {
  return (
    <View style={styles.container}>
      <Text>Account</Text>
    </View>
  );
}

export default Account;
