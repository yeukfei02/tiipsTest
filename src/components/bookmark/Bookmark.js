import React from 'react';
import { StyleSheet, View, Text } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});

function Bookmark() {
  return (
    <View style={styles.container}>
      <Text>Bookmark</Text>
    </View>
  );
}

export default Bookmark;
