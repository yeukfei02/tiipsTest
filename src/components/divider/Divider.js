import React from 'react';
import { View } from 'react-native';

function Divider(props) {
  return (
    <View style={{ margin: props.margin }} />
  );
}

export default Divider;
