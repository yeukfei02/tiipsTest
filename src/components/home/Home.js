import React from 'react';
import { StyleSheet, View, Text } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});

function Home() {
  return (
    <View style={styles.container}>
      <Text>Home</Text>
    </View>
  );
}

export default Home;
