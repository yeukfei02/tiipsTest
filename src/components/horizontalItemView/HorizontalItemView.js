import React, { useState, useEffect } from 'react';
import { StyleSheet, View, Image, Text } from 'react-native';
import { Card } from 'react-native-paper';
import { MaterialIcons } from '@expo/vector-icons';
import _ from 'lodash';
import axios from 'axios';

const ROOT_URL = `https://demo-api.bigmind.io/v1`;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 12,
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 15,
    marginBottom: 10,
    marginHorizontal: 10,
    backgroundColor: 'lightgray',
  },
  favouriteIcon: {
    alignSelf: 'flex-end'
  },
  logo: {
    width: 120,
    height: 80,
    marginTop: 10,
    alignSelf: 'center'
  },
  titleText: {
    fontSize: 13,
    fontWeight: 'bold',
    color: 'black',
    marginBottom: 5,
    alignSelf: 'center'
  },
  rowContainer: {
    flexDirection: 'row',
    alignSelf: 'center'
  },
});

function HorizontalItemView({ item }) {
  const [favouritesClicked, setFavouritesClicked] = useState(false);

  useEffect(() => {
    // getProductLogo();
  }, []);

  const getProductLogo = async () => {
    const id = item.product_id;
    const response = await axios.get(`${ROOT_URL}/products/logos/${id}`);
    console.log("response.data = ", response.data);
  }

  const renderFavouriteIcon = () => {
    let favouriteIcon = null;

    if (item) {
      if (_.isEqual(item.bookmark, "Y") || favouritesClicked) {
        favouriteIcon = (
          <MaterialIcons style={styles.favouriteIcon} name={'favorite'} size={25} color={'red'} onPress={handleFavouriteIconClick} />
        );
      } else {
        favouriteIcon = (
          <MaterialIcons style={styles.favouriteIcon} name={'favorite-border'} size={25} color={'black'} onPress={handleFavouriteIconClick} />
        );
      }
    }

    return favouriteIcon;
  }

  const handleFavouriteIconClick = () => {
    if (!favouritesClicked) {
      setFavouritesClicked(true);
    } else {
      setFavouritesClicked(false);
    }
  }

  const getClosingPrice = () => {
    let result = '';

    if (item && item.close_price) {
      result = `$${item.close_price.toFixed(3)}`;
    }

    return result;
  }

  const getPercentageChange = () => {
    let result = '';

    if (item && item.change_percentage) {
      result = `(${item.change_percentage}%)`;
    }

    return result;
  }

  return (
    <View>
      <Card style={styles.container}>
        {renderFavouriteIcon()}
        <Image
          style={styles.logo}
          source={{
            uri: 'https://reactnative.dev/img/tiny_logo.png',
          }}
        />
      </Card>
      <View>
        <Text style={styles.titleText}>{item.name}</Text>
        <View style={styles.rowContainer}>
          <Text style={{ marginRight: 5 }}>{getClosingPrice()}</Text>
          <Text>{getPercentageChange()}</Text>
        </View>
      </View>
    </View>
  );
}

export default HorizontalItemView;
