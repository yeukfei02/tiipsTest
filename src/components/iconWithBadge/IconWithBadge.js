import React from 'react';
import { View, Text } from 'react-native';
import { MaterialIcons } from '@expo/vector-icons';
import _ from 'lodash';

function IconWithBadge({ routeName, focused, size, color, badgeCount }) {
  const renderIcon = () => {
    let icon = (
      <MaterialIcons name={'favorite-border'} size={size} color={color} />
    );

    if (_.isEqual(routeName, 'Bookmark')) {
      if (focused) {
        icon = (
          <MaterialIcons name={'favorite'} size={size} color={color} />
        );
      } else {
        icon = (
          <MaterialIcons name={'favorite-border'} size={size} color={color} />
        );
      }
    }

    return icon;
  }

  return (
    <View style={{ width: 24, height: 24, margin: 5 }}>
      {renderIcon()}
      <View
        style={{
          // On React Native < 0.57 overflow outside of parent will not work on Android, see https://git.io/fhLJ8
          position: 'absolute',
          right: -6,
          top: -3,
          backgroundColor: 'red',
          borderRadius: 10,
          width: 16,
          height: 16,
          justifyContent: 'center',
          alignItems: 'center',
        }}
      >
        <Text style={{ color: 'white', fontSize: 8, fontWeight: 'bold' }}>
          {badgeCount}
        </Text>
      </View>
    </View>
  );
}

export default IconWithBadge;
