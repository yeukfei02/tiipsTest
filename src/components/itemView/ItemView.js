import React, { useState, useEffect } from 'react';
import { StyleSheet, View, Text, FlatList } from 'react-native';

import HorizontalItemView from '../horizontalItemView/HorizontalItemView';
import Divider from '../divider/Divider';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'flex-start',
    backgroundColor: 'white',
  },
  titleText: {
    fontSize: 15,
    fontWeight: 'bold',
    color: 'gray'
  }
});

function ItemView({ item }) {
  const [detailsList, setDetailsList] = useState([]);

  useEffect(() => {
    if (item) {
      setDetailsList(item.Details);
    }
  }, [item]);

  return (
    <View style={styles.container}>
      <Text style={styles.titleText}>{item.description.toUpperCase()}</Text>
      <FlatList
        data={detailsList}
        renderItem={({ item }) => {
          return (
            <HorizontalItemView
              item={item}
            />
          );
        }}
        keyExtractor={(item) => item.id}
        horizontal={true}
      />
      <Divider margin={15} />
    </View>
  );
}

export default ItemView;
