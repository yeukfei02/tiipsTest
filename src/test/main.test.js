import React from 'react';
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { shallow, mount } from "enzyme";

import Academy from '../components/academy/Academy';
import Bookmark from '../components/bookmark/Bookmark';
import Home from '../components/home/Home';
import Portfolio from '../components/portfolio/Portfolio';
import Account from '../components/account/Account';

import Divider from '../components/divider/Divider';
import IconWithBadge from '../components/iconWithBadge/IconWithBadge';
import ItemView from '../components/itemView/ItemView';
import HorizontalItemView from '../components/horizontalItemView/HorizontalItemView';

Enzyme.configure({ adapter: new Adapter() });

describe('main.test', () => {
  describe('render test', () => {
    it('Academy', () => {
      const wrapper = shallow(<Academy />);
      expect(wrapper).toMatchSnapshot();
    });

    it('Bookmark', () => {
      const wrapper = shallow(<Bookmark />);
      expect(wrapper).toMatchSnapshot();
    });

    it('Home', () => {
      const wrapper = shallow(<Home />);
      expect(wrapper).toMatchSnapshot();
    });

    it('Portfolio', () => {
      const wrapper = shallow(<Portfolio />);
      expect(wrapper).toMatchSnapshot();
    });

    it('Account', () => {
      const wrapper = shallow(<Account />);
      expect(wrapper).toMatchSnapshot();
    });

    it('Divider', () => {
      const wrapper = shallow(<Divider />);
      expect(wrapper).toMatchSnapshot();
    });

    it('IconWithBadge', () => {
      const wrapper = shallow(<IconWithBadge />);
      expect(wrapper).toMatchSnapshot();
    });

    // it('ItemView', () => {
    //   const wrapper = shallow(<ItemView />);
    //   expect(wrapper).toMatchSnapshot();
    // });
    //
    // it('HorizontalItemView', () => {
    //   const wrapper = shallow(<HorizontalItemView />);
    //   expect(wrapper).toMatchSnapshot();
    // });
  });
});
