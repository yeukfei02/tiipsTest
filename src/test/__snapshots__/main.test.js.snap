// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main.test render test Academy 1`] = `ShallowWrapper {}`;

exports[`main.test render test Account 1`] = `ShallowWrapper {}`;

exports[`main.test render test Bookmark 1`] = `ShallowWrapper {}`;

exports[`main.test render test Home 1`] = `ShallowWrapper {}`;

exports[`main.test render test Portfolio 1`] = `ShallowWrapper {}`;
